### ISSUE TITLE
DATETIME2 conversion issue

### ISSUE THREAD

**Issue by pauliev** [pymssql/pymssql#792]
**Describe the bug**
Looks like rounding issue due to treating datetime2 as datetime. 

**To Reproduce**
Define a table with time datetime2 and insert the value: '9999-12-31 23:59:59.9999999'

Once you select from table you will get an error. 

**Expected behavior**
No errors

**Current Behavior**
```
File "src/pymssql/_pymssql.pyx", line 521, in pymssql._pymssql.Cursor.fetchone
  File "src/pymssql/_pymssql.pyx", line 511, in pymssql._pymssql.Cursor.getrow
  File "src/pymssql/_mssql.pyx", line 470, in pymssql._mssql.MSSQLRowIterator.__next__
  File "src/pymssql/_mssql.pyx", line 1213, in pymssql._mssql.MSSQLConnection.fetch_next_row
  File "src/pymssql/_mssql.pyx", line 1384, in pymssql._mssql.MSSQLConnection.get_row
  File "src/pymssql/_mssql.pyx", line 875, in pymssql._mssql.MSSQLConnection.convert_db_value

ValueError: hour must be in 0..23
```

**Context (Environment)**

```
System:
platform.system()       : Linux
platform.architecture() : ('64bit', 'ELF')
platform.libc_ver()     : ('glibc', '2.31')
Python:
sys.version             : 3.8.10 (default, Nov 14 2022, 12:59:47) 
[GCC 9.4.0]
pymssql:
VERSION            : (2, 2, 7)
version            : 2.2.7
full_version       : 2.2.7
dbversion          : freetds v1.3.13
_mssql:
VERSION            : (2, 2, 7)
version            : 2.2.7
full_version       : 2.2.7
dbversion          : freetds v1.3.13
login_timeout      : 60
min_error_severity : 6
```

**Additional context**

Alright... I am not positive... my issue is hitting the datetime2 type in SQL Server. I noticed the following in `_mssql.pyx` starting around line 871

```
        elif dbtype in (SQLDATETIM4, SQLDATETIME2):
            dbconvert(self.dbproc, dbtype, data, -1, SQLDATETIME,  <--- HERE
                <BYTE *>&dt, -1)
            dbdatecrack(self.dbproc, &di, <DBDATETIME *><BYTE *>&dt)
            return datetime.datetime(di.year, di.month, di.day,
                di.hour, di.minute, di.second, di.millisecond * 1000)
```

The value `SQLDATETIME` is mapped to `SYBDATETIME` should this not be `SQLDATETIME2`

Also see https://github.com/pymssql/pymssql/issues/695

**Comment by cdquintero97** [comment]
running into a similar issue. I am using pymssql ansible module and am using a jinja template for my script. However, it is complaining about a datetime value like the op, even though the script works perfectly fine when ran directly in the sql server

**Comment by termim** [comment]
Closed in bc2cc5040ba0453bd22798bd49ed4d39ed8df3e3
